### PR TITLE
Fix Device Routes active error

### PR DESCRIPTION
**Pull request by xzzpig**
Fix When a device has multiple device routes, activating one of them will cause the others to become pending
Also added the ability to make routes inactive(why no make it to be a switch)

### CODE REVIEW

File: src/lib/common/apiFunctions.svelte [hunk 1]
```diff
@@ -267,13 +267,13 @@
...
 
-	export async function enableDeviceRoute(deviceID: string, route: string): Promise<any> {
+	export async function enableDeviceRoute(deviceID: string, routes: string[]): Promise<any> {
 		// variables in local storage
 		let headscaleURL = localStorage.getItem('headscaleURL') || '';
 		let headscaleAPIKey = localStorage.getItem('headscaleAPIKey') || '';
 
 		// endpoint url for getting users
-		let endpointURL = '/api/v1/machine/' + deviceID + '/routes?routes=' + route.replace('/', '%2F');
+		let endpointURL = '/api/v1/machine/' + deviceID + '/routes?' + routes.map(encodeURIComponent).map(route=>`routes=${route}`).join('&');
```
Comment (routerino) [inline src/lib/common/apiFunctions.svelte:276]: Ah yep, I misunderstood how this API endpoint worked, which would explain the initial issue. Also good call on the safer encode.

File: src/lib/devices/DeviceCard/DeviceRoutes.svelte [hunk 2]
```diff
@@ -39,7 +49,7 @@
 			<li>
 				{route}
 				{#if routesList.enabledRoutes.includes(route)}
-					<div class="btn btn-xs capitalize bg-success text-success-content mx-1">active</div>
+					<button on:click={() => {disableDeviceRouteAction(route)}} type="button" class="btn btn-xs capitalize bg-success text-success-content mx-1" data-tip="press to disable route">active</button>
```
Comment (routerino) [inline src/lib/devices/DeviceCard/DeviceRoutes.svelte:52]: This button needs a `tooltip` class added to use the `data-tip` functionality.

File: src/lib/devices/DeviceCard/DeviceRoutes.svelte [hunk 3]
```diff
@@ -22,7 +22,17 @@
...
+			.then((response) => {
+				getDeviceRoutesAction();
+			})
+			.catch((error) => {
+				$alertStore = error;
+			});
+  }
+
+  function disableDeviceRouteAction(route: string) {
+    enableDeviceRoute(device.id, [...routesList.enabledRoutes].filter(v=>v!=route))
```
Comment (routerino) [inline src/lib/devices/DeviceCard/DeviceRoutes.svelte:35]: What's `v=>v!=route` doing in this situation? Can you add a comment?

**Review by routerino — CHANGES_REQUESTED** [review]
looks good and makes more sense than what I was doing, just some minor fixups.

File: src/lib/devices/DeviceCard/DeviceRoutes.svelte (hunk 3 above)
Comment (xzzpig) [reply src/lib/devices/DeviceCard/DeviceRoutes.svelte:35]: The function `disableDeviceRouteAction ` needs to make the route selected by the user(`route`) to be inactive, So we need to pass the active routing list to method `enableDeviceRoute` excluding `route` by using `filter(v=>v!=route)`.
And `enableDeviceRoute(device.id, [...routesList.enabledRoutes].filter(v=>v!=route))` is same to
`var routes = [];
for(let v of routesList){
    if(v != route){
        routes.push(v)
    }
}
enableDeviceRoute(device.id, routes)
`

File: src/lib/devices/DeviceCard/DeviceRoutes.svelte (hunk 2 above)
Comment (xzzpig) [reply src/lib/devices/DeviceCard/DeviceRoutes.svelte]: Sorry, this is my mistake. I fix it in new commit.

**Review by routerino — APPROVED** [review]

**Pull request merged**